### DISCUSSION TITLE
Tasmota - Home Assistant Integration not detecting devices

### DISCUSSION THREAD

**Discussion by t-anjan** [arendst/Tasmota#12817] [Q&A] [answered]
I am using the latest version of Tasmota: 9.5.0 .

I have followed [Home Assistant - Tasmota](https://tasmota.github.io/docs/Home-Assistant/#tasmota-integration) page and set up the Tasmota integration in my Home Assistant (2021.7.4). 

* My Mosquitto (MQTT broker) is correctly configured and working.
* I have a bunch of Tasmota devices working already using the "MQTT discovery".
* I am trying to migrate one of the devices to "SetOption19 0". But the new Tasmota integration is just not detecting this device.

I have posted about this on the Home Assistant forums already ([here](https://community.home-assistant.io/t/tasmota-integration-not-detecting-devices/326083)), but I have not received any solution so far. I am hoping that the Tasmota folks here can help me with this.

(You will find a lot more logs and screenshots in the [Home Assistant forum post](https://community.home-assistant.io/t/tasmota-integration-not-detecting-devices/326083) linked above. Please do have a look.)

**Comment by barbudor** [comment, marked as answer]
Hi @t-anjan 
From your other posts in the HA forum I see that no messages relative to discovery show up in MQTT Explorer.
Do you have an ACL file in your mosquitto that may prevent the device to post to the `tasmota/discovery` topic ?

Note that I am not using HA so I have no idea how you are configuring mosquitto through HA and how ACL can be managed through HA. I am using mosquitto as a standalone docker so I know exactly what I have the configuration files.

**Reply by t-anjan** [reply]
Thank you @barbudor ! I had completely forgotten that I had set up an ACL! I had followed [this guide](https://tasmota.github.io/docs/Securing-your-IoT-from-hacking/#securing-your-communication) a long time back.

Now, I needed to add rules in the ACL to allow the device to write to the `tasmota/discovery` topic, exactly like you said.

For completeness' sake, below is a snippet of my ACL file.

```
########
######## General section
# Commented out all general rules.
# topic readwrite homeassistant/#
# topic readwrite tasmota/discovery/#
########

########
######## User section

user sonoff-living-fan-2
topic readwrite homeassistant/#
topic write tasmota/discovery/#

########

########
######## Pattern Section
# https://tasmota.github.io/docs/Securing-your-IoT-from-hacking/
pattern read cmnd/%u/#
pattern write stat/%u/#
pattern write tele/%u/#
########
```

If I could have set the mosquitto logs to "debug", I could have probably seen the "publish denied" messages in the logs. But I could not figure out how to set the mosquitto logging level in Home Assistant.

Anyway, thanks for the accurate pointer. The Tasmota discovery now works.

**Reply by barbudor** [reply]
👍

May be you could provide a doc PR to add this as a warning for other H.A. users and explain how to fix it ?

**Comment by ascillato** [comment]
Hi, can set WEBLOG to 4 and then post the output of Tasmota console when your device starts?

**Comment by MatejKovacic** [comment]
I came accross this discussion and I have similar problem. I have two Tasmota devices: Sonoff switch, which is already included in HomeAssistant. And it is working.

However, today I tried to include Tuya RGB bulb, and it is not recognized in HomeAssistant. I configured MQTT on both Tasmota devices (in Mosquitto broker I have two different users - "stikalo" (which means "switch") and "rgbzarnica" (which means "rgb bulb").

I checked Console log at devices and it seems they are connected. 

First **Sonoff switch**:
```
23:13:25.045 MQT: Attempting connection...
23:13:25.869 MQT: Connected
23:13:25.874 MQT: tele/tasmota_5C2188/LWT = Online (retained)
23:13:25.879 MQT: cmnd/tasmota_5C2188/POWER = 
23:13:27.070 MQT: stat/tasmota_5C2188/STATUS1 = {"StatusPRM":{"Baudrate":115200,"SerialConfig":"8N1","GroupTopic":"tasmotas","OtaUrl":"http://ota.tasmota.com/tasmota/release/tasmota.bin.gz","RestartReason":"Software/System restart","Uptime":"0T00:10:32","StartupUTC":"2024-04-26T22:02:55","Sleep":50,"CfgHolder":4617,"BootCount":12,"BCResetTime":"2024-04-10T15:31:03","SaveCount":153,"SaveAddress":"FB000"}}
23:13:27.156 MQT: stat/tasmota_5C2188/STATUS11 = {"StatusSTS":{"Time":"2024-04-26T23:13:27","Uptime":"0T00:10:32","UptimeSec":632,"Heap":22,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":38,"MqttCount":2,"POWER":"OFF","Wifi":***}}}
23:13:27.171 MQT: stat/tasmota_5C2188/RESULT = {"Time":"2024-04-26T23:13:27","Uptime":"0T00:10:32","UptimeSec":632,"Heap":24,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":41,"MqttCount":2,"POWER":"OFF","Wifi":{***}}
23:14:30.288 MQT: stat/tasmota_5C2188/RESULT = {"POWER":"ON"}
23:14:30.292 MQT: stat/tasmota_5C2188/POWER = ON
23:14:31.985 MQT: stat/tasmota_5C2188/RESULT = {"POWER":"OFF"}
23:14:31.989 MQT: stat/tasmota_5C2188/POWER = OFF
23:14:49.234 MQT: tele/tasmota_5C2188/STATE = {"Time":"2024-04-26T23:14:49","Uptime":"0T00:11:54","UptimeSec":714,"Heap":24,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":2,"POWER":"OFF","Wifi":{"AP":1,...}}
```

Then, **RGB bulb**:
```
23:13:25 MQT: Attempting connection...
23:13:26 MQT: Connected
23:13:26 MQT: tele/tasmota_C8B405/LWT = Online (retained)
23:13:26 MQT: cmnd/tasmota_C8B405/POWER = 
23:17:35 MQT: tele/tasmota_C8B405/STATE = {"Time":"2024-04-26T23:17:35","Uptime":"0T00:55:09","UptimeSec":3309,"Vcc":3.525,"Heap":28,"SleepMode":"Dynamic","Sleep":10,"LoadAvg":99,"MqttCount":6,"POWER":"ON","Dimmer":26,"Color":"422C0300","HSBColor":"39,96,26","White":0,"Channel":[26,17,1,0],"Scheme":0,"Fade":"OFF","Speed":1,"LedTable":"ON","Wifi":{"***}}
```

And these are the logs from Mosqutto broker:
```
2024-04-27 00:13:25: New connection from 192.168.100.165:60227 on port 1883.
2024-04-27 00:13:25: New client connected from 192.168.100.165:60227 as DVES_C8B405 (p2, c1, k30, u'rgbzarnica').
2024-04-27 00:13:26: New connection from 192.168.100.224:60962 on port 1883.
2024-04-27 00:13:26: New client connected from 192.168.100.224:60962 as DVES_5C2188 (p2, c1, k30, u'stikalo').
```

So, it looks fine, but in HomeAssistant I can see only Sonoff switch and **not** RGB bulb. Under Settings - Integrations - Tasmota, I can see only one device. Autodiscovery is on, but RGB bulb is not detected.

Any idea what could be wrong?

**Reply by MatejKovacic** [reply]
Sorry guys, false alarm. I used tasmota-lite.bin instead of tasmota.bin. Upgraded firmware via OTA, and it is working.
